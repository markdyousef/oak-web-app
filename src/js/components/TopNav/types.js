// @flow
export type DefaultProps = {
    team: null,
    router: {}
};
export type Props = {
    team: ?bool,
    logout?: Function,
    router: {
        replace: Function,
        push: Function
    },
    data: {
        loading: bool,
        me: {
            avatar: ?{
                urlThumb64: ?string
            },
            gravatar: ?string
        },
        groves: ?Array<Object>
    }
};
export type State = {
    showSettings: bool,
    showCollections: bool,
    showDialog: bool
};
