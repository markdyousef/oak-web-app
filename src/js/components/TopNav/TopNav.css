:root {
    --main-black-color: #000;
    --main-white-color: #fefeff;
    --main-orange-color: #ff9200;
    --main-red-color: #8b0000;
    --main-green-color: #005d6c;
}

.container {
    width: 100%;
    height: 60px;
    background-color: #fff;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
    position: fixed;
    z-index: 999999;
    display: flex;
    justify-content: space-between;
}

.container h1 {
    font-size: 18px;
    color: var(--main-green-color);
}

.navLeft {
    margin-left: 20px;
    height: 100%;
    display: flex;
    align-items: center;
    width: 160px;
    justify-content: space-between;
}

.navLeft img {
    height: 40px;
}

.navRight {
    height: 100%;
    align-self: flex-end;
    display: flex;
    align-items: center;
    padding-right: 30px;
}

.navRight span {
    height: 35px;
    width: 35px;
    border: 1px solid var(--main-black-color);
    border-radius: 60px;
}
