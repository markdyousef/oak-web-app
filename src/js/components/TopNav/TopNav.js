import React, { Component } from 'react';
import { Link } from 'react-router';

import css from './TopNav.css';
import logo from '../../../img/full_blue.png';
import Avatar from '../shared/Avatar';

const IMG = '//style.anu.edu.au/_anu/4/images/placeholders/person.png'
class TopNav extends Component {
    constructor() {
        super();
        this.state = {};
    }
    render() {
        return (
            <nav className={css.container}>
                <div className={css.navLeft}>
                    <img src={logo} alt="logo" />
                    <h1>MASTERMIND</h1>
                </div>
                <div className={css.navRight}>
                    <Link to="/">
                        <Avatar img={IMG} />
                    </Link>
                </div>
            </nav>
        );
    }
}

export default TopNav;
