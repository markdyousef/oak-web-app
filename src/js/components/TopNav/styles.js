import styled from 'styled-components';
import colors from '../../styles/colors';

export const Container = styled.nav`
    width: 100%;
    height: 60px;
    background-color: #fff;
    border-bottom: 1px solid ${colors.lightGrey};
    display: flex;
    justify-content: space-between;
    position: relative;
    flex-shrink: 0;
`;

export const Profile = styled.button`
    height: 32px;
    width: 32px;
    border-radius: 999em;
    border: 1px solid #E5E5E5;
    margin-left: 5px;
    padding: 0;
    margin-left: 15px;
`;

export const NavRight = styled.div`
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: flex-end;
    margin-right: 25px;
    width: 200px;
`;

export const NavLeft = styled.div`
    margin-left: 25px;
    height: 100%;
    display: flex;
    align-items: center;
    width: 200px;
`;

export const NavCenter = styled.div`
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
`;

export const Dropdown = styled.div`
    top: 60px;
    position: absolute;
    min-width: 200px;
    z-index: 9999;
`;

export const Collections = styled.button`
    padding: 0;
    border: 0;
    background-color: #fff;
    font-size: 15px;
    margin-left: 7px;
    font-family: 'Proxima-Nova';
    cursor: pointer;
`;

export const Item = styled.div`
    width: 100%;
    cursor: pointer;
    padding-top: 5px;
    margin-top: 10px;
`;

export const ItemTitle = styled.div`
    color: ${colors.grey};
    font-size: 12px;
`;

export const All = styled.div`
    padding: 16px 0;
    color: ${colors.grey};
    font-size: 15px;
    border-bottom: 1px solid ${colors.lightGrey};
`;

export const Add = styled.div`
    display: flex;
    margin-top: 16px;
`;
