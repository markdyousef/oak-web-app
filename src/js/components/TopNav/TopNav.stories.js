import React from 'react';
import { storiesOf } from '@kadira/storybook';
import TopNav from './TopNav';

storiesOf('TopNav', module)
    .add('empty', () => (
        <TopNav />
    ));
