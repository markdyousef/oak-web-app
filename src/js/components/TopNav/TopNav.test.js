// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import TopNav from './TopNav';

test('renders TopNav', () => {
    const props = {
        data: {}
    };
    const component = renderer.create(
        <TopNav {...props} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
