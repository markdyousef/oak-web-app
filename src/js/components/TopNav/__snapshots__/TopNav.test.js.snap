exports[`test renders TopNav 1`] = `
<nav
  className={undefined}>
  <div
    className={undefined}>
    <img
      alt="logo"
      src="test-file-stub" />
  </div>
  <div
    className={undefined}>
    <a
      onClick={[Function]}
      style={Object {}}>
      TEAMS
    </a>
    <a
      className={undefined}
      onClick={[Function]}
      style={Object {}}>
      <div
        className={undefined}>
        <img
          alt="profile"
          src="//style.anu.edu.au/_anu/4/images/placeholders/person.png" />
      </div>
    </a>
  </div>
</nav>
`;
